### PR TITLE
Feature/milestone3

### PR DESCRIPTION
**Pull request by gonjavi**
Requirements
- Make the BooksForm component stateful
The component should store the title and the book category in its own state.
The component should implement the handleChange method which reacts to the user typing to input a name or to changes in the category select.
Each change to the form should modify the component's state.
The component should implement handleSubmit which will save the new book in the Redux store and resets the component's state.
Documentation: Map Dispatch to Props
- Connect the REMOVE_BOOK action to the component
Add a new table column to the BooksList component. It should render the "Remove Book" button next to each book.
Implement the handleRemoveBook method which accepts a book as an argument and removes it from the Redux store.

### CODE REVIEW

**Review by fatymahmed — CHANGES_REQUESTED** [review]
## Hi @gonjavi,
you have done a great job in this milestone 👏👏
Please make the following changes for it to be approved
- Avoid disabling linter rules, see my comments below for a detailed review regarding this.
- Remember that you should implement only one `handleChange` method then attach it to `input's` and `select's` `onChange` event, see [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone3/HandleChange.png) from pathwright
- Add more information regarding the project to your readme file

### Status: Changes Required ♻️
_Feel free to reach out to me on slack (Fatima) if you have any questions or need any help🙂_

File: src/containers/BookForm.js [hunk 1]
```diff
@@ -1,39 +1,69 @@
...
+      title: e.target.value,
+      id: (Math.random() * 500).toFixed(),
+    });
+  }
+
+  handleSelect(evt) {
+    this.setState({
+      category: evt,
+    });
+  }
```
Comment (fatymahmed) [inline src/containers/BookForm.js]: - Combine these 2 methods to be one method as mentioned in the requirements, see [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone3/HandleChange.png)

you could do something like this

`handleChange(event) {
  const name = event.target.name;
  this.setState({
    title: event.target.value
    [name]: event.target.value
  })
}`

File: src/containers/BookForm.js [hunk 2]
```diff
@@ -1,39 +1,69 @@
...
+
+  handleSubmit(e) {
+    e.preventDefault();
+    const book = this.state;
+    // eslint-disable-next-line react/destructuring-assignment
+    this.props.createBook(book);
```
Comment (fatymahmed) [inline src/containers/BookForm.js]: - avoid disabling this, it's easy to fix this rule, you should destructure it then use the method.

File: src/containers/BooksList.js [hunk 3]
```diff
@@ -1,27 +1,55 @@
...
+class BooksList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleRemove = this.handleRemove.bind(this);
+  }
+
+  handleRemove(id) {
+    // eslint-disable-next-line react/destructuring-assignment
+    this.props.removeBook(id);
+  }
```
Comment (fatymahmed) [inline src/containers/BooksList.js]: - avoid disabling this, it's easy to fix this rule, you should destructure it then use the method.

File: src/containers/BooksList.js [hunk 4]
```diff
@@ -1,27 +1,55 @@
...
+
+  render() {
+    const { books } = this.props;
+    const BooksList = books.books.map(
+      // eslint-disable-next-line max-len
+      b => <Book key={b.id} Id={b.id} title={b.title} cat={b.category} Click={() => this.handleRemove(b.id)} />,
```
Comment (fatymahmed) [inline src/containers/BooksList.js]: - You could have this statement in multiple lines to solve this , no need to disable it

File: src/containers/BookForm.js (hunk 1 above)
Comment (gonjavi) [reply src/containers/BookForm.js]: Ok thanks

**Review by bellom — APPROVED** [review]
## Hi @gonjavi  :clap: 

**Great work making all the necessary corrections from the previous review.**

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit project completion form.
Good luck in the next milestone :slightly_smiling_face:_ 

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Pull request merged**